src/Workspaces/Core/Desktop/Workspace/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -192,6 +192,8 @@ private static Task<MSB.Execution.BuildResult> BuildAsync(MSB.Execution.BuildMan
                         taskSource.TrySetCanceled();
                     }
                 });
+
+                taskSource.Task.SafeContinueWith(_ => registration.Dispose(), CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
```

Comment:
i'm not a huge fan of hte SafeContinueWith approach.  Can we instead just call an async method that awaits taskSource, then calls registrtion.Dispose in a finally?